kratos/solving_strategies/convergencecriterias/displacement_criteria.h

```diff
@@ -424,21 +417,22 @@ class DisplacementCriteria
         TDataType final_correction_norm = TDataType();
         SizeType dof_num = 0;
 
+        // Custom reduction
+        using CustomReduction = CombinedReduction<SumReduction<TDataType>,SumReduction<SizeType>>;
+
+        // Auxiliary struct
+        struct TLS {TDataType dof_value{};IndexType dof_id{}; TDataType variation_dof_value{};};
+
         // Loop over Dofs
-        #pragma omp parallel for reduction(+:final_correction_norm,dof_num)
-        for (int i = 0; i < static_cast<int>(rDofSet.size()); i++) {
-            auto it_dof = rDofSet.begin() + i;
-
-            IndexType dof_id;
-            TDataType variation_dof_value;
-
-            if (it_dof->IsFree()) {
-                dof_id = it_dof->EquationId();
-                variation_dof_value = Dx[dof_id];
-                final_correction_norm += std::pow(variation_dof_value, 2);
-                dof_num++;
+        std::tie(final_correction_norm, dof_num) = block_for_each<CustomReduction>(rDofSet, TLS(), [&Dx](auto& rDof, TLS& rTLS) {
+            if (rDof.IsFree()) {
+                rTLS.dof_id = rDof.EquationId();
+                rTLS.variation_dof_value = Dx[rTLS.dof_id];
+                return std::make_tuple(std::pow(rTLS.variation_dof_value, 2), 1);
+            } else {
+                return std::make_tuple(0.0, 0);
```

Comment:
For consistency, better to return `std::make_tuple(TDataType(), 0);` (I like `0.0` the better, but since the original code had this probably better to respect it)

Reply:
Okidoki